### PR TITLE
C++: Fix characteristic predicate of `XMLParent`.

### PR DESCRIPTION
**Pull request by geoffw0**
This is the C++ version of https://github.com/Semmle/ql/pull/2520 / https://github.com/Semmle/ql/pull/2529.

I've also added a test - we don't actually have a full test for XML.qll (it wouldn't be trivial to write) so I've added a simple regression test for this issue in the files test instead for now.  And I've modernized the files test.

### CODE REVIEW

File: cpp/ql/test/library-tests/files/Files.ql [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+string describe(File f) {
+  f.compiledAsC() and
+  result = "C"
+  or
+  f.compiledAsCpp() and
+  result = "C++"
+  or
+  f instanceof XMLParent and
+  result = "XMLParent" // regression tests a bug in the constructor of XMLParent
```
Comment (max-schaefer) [inline cpp/ql/test/library-tests/files/Files.ql]:
```suggestion
  result = "XMLParent" // regression tests a bug in the characteristic predicate of XMLParent
```

Comment (max-schaefer) [reply cpp/ql/test/library-tests/files/Files.ql]: This case doesn't actually seem to be exercised by the test?

Comment (geoffw0) [reply cpp/ql/test/library-tests/files/Files.ql]: It is in the second commit, then the third (rightly) removed those results.  So if we ever regressed the char pred of XMLParent, those results would show up again.

Comment (max-schaefer) [reply cpp/ql/test/library-tests/files/Files.ql]: Makes sense.

**Comment by max-schaefer** [comment]
LGTM. @jbj, do you want to take a look as well, or should I merge?

**Pull request merged**